testinterface: specify reduced backlog for ServerSocket

LeeTibbert opened the pull request:
* Issue #1585 described a number of opportunities to improve
    the SN testinterface.  A number of these changes are in the
    same file: ComRunner.scala.  I believe that review & merge
    will be easier if the changes are submitted as a series of PRs.

  * This PR fixes a small, easily identifiable change, suitable to
    the small amount of time I had available. If the backlog
    parameter (number of possible/expected inbound connections) is
    not specified when the ServerSocket is instantiated, it defaults
    to 50.  It is clear that only one inbound connection is ever
    expected.  Specifying that expectation reduces the amount of
    resources used at that point, and quickly freed.

    This is not the biggest, nor most important improvement but it
    is the direction of progress and reduces cognitive overhead
    whilst the dreaded "EOF handshake" nest of bugs is addressed.

#### Documentation:

  * The affected file is internal to the Scala Native build itself.
    No documentation change needed.

#### Testing:

###### Safety
  * Built ("rebuild")and tested ("test-all") in debug mode using sbt 1.3.10
    & Java 8 on X86_64 only . All tests pass.
###### Efficacy
  * No cost effective way to test for efficacy.

LeeTibbert commented:
Normally I would have written `new ServerSocket(0, backlog = 1)`  so that someone reading the code did not have to dive into the socket API to make sense of the call.  Unfortunately, the constructor is private, so the name of the argument is not available (well, other than a x$2, which detracts rather than adding to clarity).  I checked this against Scala JVM (scalastie).  If I missed something and it is possible to name the argument, please advise.

sjrd commented on test-runner/src/main/scala/scala/scalanative/testinterface/ComRunner.scala:
```diff
@@ -35,7 +35,7 @@ class ComRunner(bin: File,
   private[this] val socket =
     try {
-      serverSocket = new ServerSocket(0)
+      serverSocket = new ServerSocket(0, 1)
```
Consider using comment-named params like this:
```suggestion
      serverSocket = new ServerSocket(/* port = */ 0, /* backlog = */ 1)
```

LeeTibbert replied:
Done. Good approach! Thank you.  I will have to add that idiom to my tool bag.
Scalafmt instisted on a space after the open-paren.

LeeTibbert commented:
My thanks to you both for the timely review & merge.  
I have am studying and addressing the review comments in two higher
priority PRs, but have already sandboxed two more PRs in this series.
Everything in time....
